src/index.ts

```diff
@@ -1 +1,20 @@
-export { default as BeaconDetector } from './BeaconDetector';
+import os from 'os';
+
+import logger from './logger';
+
+export const beaconDetectorResolver = new Promise(async $export => {
+    let module;
+
+    switch (os.type()) {
```

Comment:
I've always seen checks like this performed with https://nodejs.org/api/process.html#process_process_platform not sure if there is any important difference.